plugins/prometheus.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*-
+from jinja2 import Template
+import json
+from matrix_client.api import MatrixRequestError
+from neb.engine import KeyValueStore, RoomContextStore
+from neb.plugins import Plugin, admin_only
+from Queue import PriorityQueue
+from threading import Thread
+
+
+import time
+import logging as log
+import random
+
+queue = PriorityQueue()
+
+
+class PrometheusPlugin(Plugin):
+    """Plugin for interacting with Prometheus.
+    """
+    name = "prometheus"
+
+    #Webhooks:
+        #    /neb/prometheus
+    TYPE_TRACK = "org.matrix.neb.plugin.prometheus.projects.tracking"
+
+    TRACKING = ["track", "tracking"]
+
+    def __init__(self, *args, **kwargs):
+        super(PrometheusPlugin, self).__init__(*args, **kwargs)
+        self.store = KeyValueStore("prometheus.json")
+        self.rooms = RoomContextStore(
+            [PrometheusPlugin.TYPE_TRACK]
+        )
+        self.queue_counter = 1L
+        self.consumer = MessageConsumer(self.matrix)
+        self.consumer.daemon = True
+        self.consumer.start()
+
+    def on_receive_message(self, info):
```

Comment:
I would be tempted to fold this function into `on_receive_webhook`. The `Plugin` class has a number of "over-ridable" functions but this is *not* one of them. Given this function isn't huge, moving it to `on_receive_webhook` makes sense.